azure/durable_functions/models/DurableOrchestrationContext.py

```diff
@@ -684,7 +684,7 @@ def _add_to_open_tasks(self, task: TaskBase):
                 task.id = self._sequence_number
                 self._sequence_number += 1
                 self.open_tasks[task.id] = task
-            elif task.id != -1:
+            elif task.id != -1 and self.open_tasks[task.id] != task:
                 self.open_tasks[task.id].append(task)
```

Comment:
(don't block on my feedback)
It's been a bit since I reviewed this codepath, but I seem to recall that this `elif` statement is specifically for **external events**.

For example, if you do:

```python
t1 = context.wait_for_external_event("A")
t2 = context.wait_for_external_event("A")
t = context.when_all([t1,t2])
yield t
```

then in `yield t` you should end up with `open_tasks` as:

```json
{
"A": [t1, t2]
}
```

I don't think the edit to this `elif` maintains that guarantee. Or does it?
For this change, please use my external events snippet above to ensure this functionality has not regressed.

Reply:
Oh wait, it definitely does. Sorry, misread. Feel free to dismiss.

Reply:
Yep! Tested it with this case, the check passes and the code executes as it would have in this case. I'll add a comment here so we can remember what this check is for in future :)